### PR TITLE
refactor how mirror data is handled

### PR DESCRIPTION
**Pull request by mwhudson**
This is prep work for allowing autoinstalls to specify a complete apt
config.

### CODE REVIEW

File: subiquity/models/mirror.py [hunk 1]
```diff
@@ -13,35 +13,61 @@
...
+        {
+            "arches": ["default"],
+            "uri": "http://ports.ubuntu.com/ubuntu-ports",
+        },
+        ],
+}
+
+
+def get_default_mirror():
+    return get_mirror(DEFAULT, "primary", get_architecture())
```
Comment (raharper) [inline subiquity/models/mirror.py]: Could we move the get_mirror() call to the object initialization so we can cache it?  It won't change during the lifetime of the install so we can save some execs and overhead if we keep it around.

Comment (raharper) [reply subiquity/models/mirror.py]: Alternatively, we recently made use of python3's lru_cache() in cloud-init

from functools import lru_cache()

@lru_cache()
def get_default_mirror()

...

File: subiquity/models/mirror.py [hunk 2]
```diff
@@ -13,35 +13,61 @@
...
 
     def __init__(self):
-        self.mirror = DEFAULT_MIRROR
+        self.config = copy.deepcopy(DEFAULT)
 
     def set_country(self, cc):
-        parsed = parse.urlparse(DEFAULT_MIRROR)
+        uri = self.get_mirror()
+        if uri != get_default_mirror():
+            return
```
Comment (raharper) [inline subiquity/models/mirror.py]: Why do we return if the mirror and default mirror don't match?

Comment (mwhudson) [reply subiquity/models/mirror.py]: This is looking ahead to autoinstall: if the user sets the mirror to http://my-private-mirror.internal we don't want to rewrite that to http://CC.my-private-mirror.internal.

File: subiquity/models/mirror.py (hunk 1 above)
Comment (mwhudson) [reply subiquity/models/mirror.py]: Tweaked a bit, see what you think now.

File: subiquity/models/mirror.py [hunk 3]
```diff
@@ -13,35 +13,59 @@
...
+            "uri": "http://archive.ubuntu.com/ubuntu",
+        },
+        {
+            "arches": ["default"],
+            "uri": "http://ports.ubuntu.com/ubuntu-ports",
+        },
+        ],
+}
+
+ARCHITECTURE = get_architecture()
```
Comment (raharper) [inline subiquity/models/mirror.py]: Doesn't this create a side-effect of executing dpkg if something imports this module?

Comment (raharper) [reply subiquity/models/mirror.py]: AFAICT, nothing outside of this module uses DEFAULT_MIRROR or ARCHITECTURE and the API being used are the method son the MirrorModel itself, could we move these values to the class itself? then they can be properties that are cached after initialization in __init__?

Comment (mwhudson) [reply subiquity/models/mirror.py]: Yeah, that makes more sense, thanks.

**Pull request merged**